turbinia/client.py

```diff
@@ -710,6 +731,90 @@ def format_task_statistics(
     report.append('')
     return '\n'.join(report)
 
+  def format_worker_status(
+      self, instance, project, region, days=0, all_fields=False):
+    """Formats the recent history for Turbinia Workers.
+
+    Args:
+      instance (string): The Turbinia instance name (by default the same as the
+          INSTANCE_ID in the config).
+      project (string): The name of the project.
+      region (string): The name of the zone to execute in.
+      days (int): The number of days we want history for.
+      all_fields (bool): Include historical Task information for the worker.
+    Returns:
+      String of Request status
+    """
+    # Set number of days to retrieve data
+    num_days = 7
+    if days != 0:
+      num_days = days
+    task_results = self.get_task_data(instance, project, region, days=num_days)
+    if not task_results:
+      return ''
+
+    # Sort task_results by last updated timestamp.
+    task_results = sorted(
+        task_results, key=itemgetter('last_update'), reverse=True)
+
+    # Create dictionary of worker_node: {{task_id, task_update,
+    # task_name, task_status}}
+    workers_dict = {}
+    for result in task_results:
+      worker_node = result.get('worker_name')
+      status = result.get('status')
+      status = status if status else 'No task status'
+      if worker_node and worker_node not in workers_dict:
+        workers_dict[worker_node] = []
+      if worker_node:
+        task_dict = {}
+        task_dict['task_id'] = result.get('id')
+        task_dict['last_update'] = result.get('last_update')
+        task_dict['task_name'] = result.get('name')
+        task_dict['status'] = status
+        # Check status for anything that is running.
+        if 'running' in status:
+          run_time = (datetime.now() -
+                      result.get('last_update')).total_seconds()
+          run_time = timedelta(seconds=run_time)
+          task_dict['run_time'] = run_time
+        else:
+          task_dict['run_time'] = result.get('run_time')
```

Comment:
Won't this fail if the tasks are queued?  Maybe just check for `run_time` instead?

Reply:
So it will just return `None` if it doesn't exist, but I updated it to check if its `None` and change the status to `No run time.` to make it more clear to the user